src/library/browse/foldertreemodel.cpp

```diff
@@ -1,28 +1,95 @@
-#if defined (__WINDOWS__)
+#if defined(__WINDOWS__)
 #include <windows.h>
 #include <Shellapi.h>
 #include <Shlobj.h>
 #else
-#include <sys/types.h>
-#include <sys/stat.h>
 #include <dirent.h>
-#include <unistd.h>
 #include <errno.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
 #endif
 
 #include <QFileInfo>
+#include <QtConcurrent>
 
 #include "library/browse/browsefeature.h"
 #include "library/browse/foldertreemodel.h"
 #include "library/treeitem.h"
 #include "moc_foldertreemodel.cpp"
 #include "util/file.h"
+#include "util/qt.h"
+
+FolderTreeModel::FolderTreeModel(QObject* parent)
+        : TreeItemModel(parent), m_isRunning(true) {
+    QObject::connect(&m_fsWatcher,
+            SIGNAL(directoryChanged(QString)),
+            this,
+            SLOT(dirModified(QString)));
+    connect(this,
+            &FolderTreeModel::newChildren,
+            this,
+            &FolderTreeModel::addChildren,
+            Qt::QueuedConnection);
```

Comment:
```suggestion
    connect(this,
            &FolderTreeModel::newChildren,
            this,
            &FolderTreeModel::addChildren);
```
@uklotzde for example something like this, what you had in your mind?

because unfortunately in this case i'm getting:

```
warning [Thread (pooled)] QObject::connect: Cannot queue arguments of type 'TreeItemList'
(Make sure 'TreeItemList' is registered using qRegisterMetaType().)
```

even though it is being registered here:
https://github.com/mixxxdj/mixxx/pull/3623/files#diff-6513e9ba2e9d2f7199ec516b9e08cc168423198f8177e411d1053acbfb1c5ef2R72

and this makes the whole thing break...